pxtsim/simdriver.ts

```diff
@@ -314,7 +315,7 @@ namespace pxsim {
                 const single = !!this._currentRuntime?.single;
                 const parentWindow = window.parent && window.parent !== window.window
                     ? window.parent : window.opener;
-                if (this.options.nestedEditorSim && parentWindow) {
+                if (parentWindow) {
```

Comment:
After taking another look I think we might want to separate this out from the two connected 'else if's that follow now that it's not limited to `nestedEditorSim` -- otherwise those won't trigger in any embed scenario. So just leaving this if as is & having line 325 cut out the 'else' ~

Reply:
When remove the else on 325, I get multiple simulators instanced. I think I need a better understanding of this function.

Reply:
Ahh gotcha, I see. We were kinda bypassing an issue with how it was set up by having it in the same branch.

This issue is that broadcast messages we use to send multiplayer state are often used to represent e.g. radio, so it falls into the the path below where it says "oh you want to simulate something that communicates with other boards? Okay, make another sim".

So, to get that to work nicely, we should probably do two things -- add an optional param to SimulatorBroadCastMessage (https://github.com/microsoft/pxt/blob/viewBtnControls/pxtsim/embed.ts#L81) for 'toParentIFrameOnly' which we'll set to true in the call to `Runtime.postMessage` (in `pxt-common-packages/libs/game/sim/multiplayer.ts`), and wrap the second two branches that start on 325 of this in another `if (!this.options.nestedEditorSim && !broadcastmsg?.toParentIFrameOnly)` to avoid double sim when we only really want to post to the outer frame